### PR TITLE
[TD/RA] Move some files to common library

### PR DESCRIPTION
**Pull request by wdigger**

### CODE REVIEW

**Comment by hifi** [comment]
Nice work :ok_hand:

File: CMakeLists.txt [hunk 1]
```diff
@@ -64,8 +64,40 @@ add_library(common STATIC
...
+    common/timerini.cpp
+    common/windows.cpp
+    common/winhide.cpp
+    common/writepcx.cpp
+    common/delay.cpp
 )
 
+file(GLOB_RECURSE COMMON_HEADERS "common/*.h")
+
+add_library(common STATIC ${COMMON_SRC} ${COMMON_HEADERS})
```
Comment (OmniBlade) [inline CMakeLists.txt:97]: With the inclusion of gbuffer and ddraw, you will need to built common twice, once as common with REMASTER_BUILD set without linking to the ddraw libs and once without it set, so just have it build commonv from the sources as well and link all the libraries just for that one.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (wdigger) [reply CMakeLists.txt:97]: I removed SetVideoMode calls from RE.
DDraw not linking with dlls anymore, at least in release.

Comment (OmniBlade) [reply CMakeLists.txt:97]: I think for future proofing its probably a good idea to separate the vanilla and remaster builds of common completely so we have more freedom to add code the differs between remasters and standalone builds. Something like the following:

```CMake
file(GLOB_RECURSE COMMON_HEADERS "common/*.h")

if(BUILD_REMASTERTD OR BUILD_REMASTERRA) 
    add_library(common STATIC ${COMMON_SRC} ${COMMON_HEADERS})
    target_link_libraries(common PUBLIC winmm)
    target_compile_definitions(common PUBLIC REMASTER_BUILD)
endif()

if(BUILD_VANILLATD OR BUILD_VANILLARA)
    add_library(commonv STATIC
        ${COMMON_SRC}
        ${COMMON_HEADERS}
        common/soundio.cpp
    )

    target_link_libraries(commonv PUBLIC ddraw dxguid dsound winmm)
endif()
```

Comment (wdigger) [reply CMakeLists.txt:97]: I believe that renaster not really interested in video, sound and input, it is possible we can separate logic at all. Future will tell.

Comment (wdigger) [reply CMakeLists.txt:97]: Other way - extract interface for vide/sound/input and make different realizations for different cases.
video_ddraw.lib
video_sdl.lib
video_dummy.lib
and so on...

Comment (OmniBlade) [reply CMakeLists.txt:97]: Can you move ddraw and dxguid as link libraries to commonv only then so we remove the risk of accidentally linking against them when we don't intend to? We can look at creating separate implementations for video and other platform specific bits as we work on porting the game to another framework.

Comment (wdigger) [reply CMakeLists.txt:97]: Nop, right now ddraw uses in many places across all the library code. I believe this big work is not in scope of this PR.

Comment (hifi) [reply CMakeLists.txt:97]: I don't think the remaster code needs to link ddraw, it only uses headers and data types.

Comment (wdigger) [reply CMakeLists.txt:97]: Remaster release don't linked with ddraw.

Comment (OmniBlade) [reply CMakeLists.txt:97]: Which is why common shouldn't be linking it either, so there is no chance of accidentally pulling it in by calling something in remaster code down the line unintentionally. Anyhow, latest changes look okay to me.

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**